File: C++/primes.cc
#include <cmath>
#include <cstdio>
#include <cstdlib>
int main(int, char **argv) {
	printf("2");
	unsigned limit = atoi(argv[1]), half = limit / 2, total = 1,
			 isqrt = sqrt(limit);
	auto sieve = new bool[half]();
	for (auto i = 3u; i <= isqrt; i += 2)
		if (!sieve[i / 2])
			for (auto j = i * i / 2; j < half; j += i) sieve[j] = true;
	for (auto i = 1u; i < half; i++)
		if (!sieve[i]) {
			printf(", %u", 2 * i + 1);
			total++;
		}
	printf("\nTotal: %u\n", total);
}

File: Rust/primes.rs
fn main() {
	print!("2");
	let limit = std::env::args().collect::<Vec<String>>()[1]
		.parse::<usize>()
		.unwrap();
	let half = limit / 2;
	let (mut sieve, mut total) = (vec![true; half], 1);
	for i in (3..=(limit as f32).sqrt() as usize).step_by(2) {
		if sieve[i / 2] {
			for j in (i * i / 2..half).step_by(i) {
				sieve[j] = false;
			}
		}
	}
	for i in 1..half {
		if sieve[i] {
			print!(", {}", 2 * i + 1);
			total += 1;
		}
	}
	println!("\nTotal: {}", total);
}
